src/toolbox.ts

```diff
@@ -1999,56 +2017,36 @@ export class FilterPointDistanceFromRow extends ToolboxItem {
         this.config = this.ulabel.config.distance_filter_toolbox_item
 
-        // Create a set of defaults for every config value
-        const default_values = {
-            "name": <string> "Filter Distance From Row",
-            "component_name": <string> "fitler_distance_from_row",
-            "filter_min": <number> 0,
-            "filter_max": <number> 100,
-            "default_values": <AnnotationClassDistanceData> {"single": 50},
-            "step_value": <number> 1,
-            "multi_class_mode": <boolean> false,
-            "filter_on_load": <boolean> true,
-            "show_options": <boolean> true,
-            "toggle_overlay_keybind": <string> "p"
-        }
-
-        // Loop through every key value pair in the config
-        for (const [key, value] of Object.entries(this.config)) {
-            // If the passed in value's type !== the default value's type then use the default value
-            if (typeof value !== typeof default_values[key]) {
-                this.config[key] = default_values[key]
+        // For each key missing from the config, set the default value
+        for (const key in DEFAULT_FILTER_DISTANCE_CONFIG) {
+            if (!this.config.hasOwnProperty(key)) {
+                this.config[key] = DEFAULT_FILTER_DISTANCE_CONFIG[key]
             }
         }
 
         // Set the component's properties to be the same as the config's properties
         for (const property in this.config) {
             this[property] = this.config[property]
         }
+
+        // Force disable multi-class mode if the config doesn't allow it
+        if (this.disable_multi_class_mode) this.multi_class_mode = false
         
         // Get if the options should be collapsed from local storage
-        if (window.localStorage.getItem("filterDistanceCollapseOptions") === "true") {
-            this.collapse_options = true
-        }
-        else if (window.localStorage.getItem("filterDistanceCollapseOptions") === "false") {
-            this.collapse_options = false
-        }
+        this.collapse_options = window.localStorage.getItem("filterDistanceCollapseOptions") === "true"
  
         // Create an overlay and determine whether or not it should be displayed
         this.create_overlay()
-        if (window.localStorage.getItem("filterDistanceShowOverlay") === "true") {
-            this.show_overlay = true
-            this.overlay.update_display_overlay(true)
-        }
-        else if (window.localStorage.getItem("filterDistanceShowOverlay") === "false") {
-            this.show_overlay = false
-            this.overlay.update_display_overlay(false)
-        }
-        else if (this.config.show_overlay_on_load !== undefined || this.config.show_overlay_on_load !== null) {
-            this.show_overlay = this.config.show_overlay_on_load
+
+        // Check if localStorage has a value for showing the overlay
+        if (window.localStorage.getItem("filterDistanceShowOverlay") !== null) {
```

Comment:
Would it be worth pulling these into a separate interface with types so we don't need to access it with strings?
We only use it here in `toolbox.ts`, but it does have a fair amount of calls. 
Also should probably standardize on `window.localStorage` vs `localStorage`.

Reply:
[fd70506](https://github.com/SenteraLLC/ulabel/pull/167/commits/fd70506e3335b0d5cf55d65e85ac0818041402a8)